puppet/fabfile.py

```diff
@@ -128,6 +170,11 @@ def puppet_apply(dry_run='no'):
             " "+manifest_to_run+" "
             )
 
```

Comment:
this is pretty much on the money.

There's an edge case I can see which is, the 'puppet apply' starts ubersystem (via supervisord) when it's finished running.    After this we then upgrade the DB, but not restarting the process, so if something requires the DB to be migrated for uber to start it up, nothing here will restart it.  The email thread kicks off right at the start so it could spam errors until someone manually restarts or does another deploy.

The quick fix is simply to add ```restart_uber_service()``` right after ```upgrade_db()```.

The probably more correct (but maybe slightly more involved) fix is to have puppet not automatically start the service, and then surround this function with ```stop_uber_service()``` and ```start_uber_service()``` so that everything happens while sideboard isn't started.

I -think- what we'd do is modify our puppet declaration of supervisor here: https://github.com/magfest/ubersystem-puppet/blob/master/manifests/daemon.pp#L11   and change it from ensuring that it starts automatically, to making it not start automatically. 

Unfortunately it looks like our version of the puppet supervisor scripts are old, and we'd also need to upgrade them from latest version.  Doing that (which isn't too hard, mostly just cloning the github repo into the right dir) I think would mean we would just add ```ensure_process => 'stopped'``` and that would fix it.

newer version of the file (we'd have to upgrade the entire module) https://github.com/ajcrowe/puppet-supervisord/blob/master/manifests/program.pp

Doing the better fix here is probably a good idea anyway because (I haven't verified this) but I think somewhere along the lines, ubersystem is not being shut off prior to starting the deploy, so it's autorestarting as puppet is updating the git files, occasionally causing weirdness.

Let's sync up probably make this version of it happen.